docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+                              "SALES", 
+                              new FileEndpoint("/tmp/output", 
+                                               "out.file", 
+                                               new CSVMessageFormat(<>))
+                            );
+```
+
+**_"registerTable"_** API takes the name of the table and an instance of endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for the data. Each of these endpoints requires a different type of configurations and possibly data formats. In this example, data format is defined using _MessageFormat_ by specifying it at the endpoint creation. _CSVMessageFormat_ can be configured with schema string as follows:
+```sql
+{
+  "separator": ",",
+  "quoteChar": "\"",
+  "fields": [
+    {
+      "name": "RowTime",
+      "type": "Date",
+      "constraints": {
+        "format": "dd/MM/yyyy hh:mm:ss Z"
+      }
+    },
+    {
+      "name": "id",
+      "type": "Integer"
+    },
+    {
+      "name": "Product",
+      "type": "String"
+    },
+    {
+      "name": "units",
+      "type": "Integer"
+    }
+  ]
+}
+```
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Following data endpoints are supported: 
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
+* **FileEndpoint**
+: It needs to be configured with directory path, file name and MessageFormat as in line 2 in above code 
+* **StreamEndpoint** 
+: This allows us to connect existing operator output or input to SQL query as a data source or sink respectively. StreamEndpoint needs immediate downstream operator's input port or immediate upstream operator's output port and field mapping for CSV data or POJO class. This will be explained in detail in next [example](#fusion-style-sql-application).
+
+
+### Using User Defined functions ( UDF ) in SQL query
+
+We can use our own scalar UDF, implemented in Java, in a SQL statement for data manipulation. But before the usage, we need to register the function with execution environment by using registerFunction API.
+
+```java
+sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str");
+```
+
+In above code, **_registerFunction_** takes UDF name to be used in SQL, JAVA class which implements static method and name of the same method as parameters. 
+Used static method "apex_concat_str" takes two String objects as input parameters as a requirement from SQL query.
+```java    
+public static String apex_concat_str(String s1, String s2)
+{
+  return s1 + s2;
+}
+```
+
+The scalar UDF "APEXCONCAT" can be used in SQL as described below whereas FLOOR & SUBSTRING are standard SQL scalar functions supported by Apache Calcite.
```

Comment:
can be => that was registered above can be
below whereas FLOOR & SUBSTRING => below. FLOOR and SUBSTRING